x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -57,10 +63,83 @@ export const shorthandMap = {
   },
 };
 
+export const hasReadIndexPrivileges = async (
+  privileges: Privilege,
+  logger: Logger,
+  buildRuleMessage: BuildRuleMessage,
+  ruleStatusService: RuleStatusService
+): Promise<boolean> => {
+  const indexNames = Object.keys(privileges.index);
+  const [indexesWithReadPrivileges, indexesWithNoReadPrivileges] = partition(
+    indexNames,
+    (indexName) => privileges.index[indexName].read
+  );
+
+  let errorString: string | null = null;
+
+  if (indexesWithReadPrivileges.length > 0 && indexesWithNoReadPrivileges.length > 0) {
```

Comment:
Super nit: could probably move these checks to a separate function that return an error string or null to avoid the `let` here.

Reply:
Even abstracting away the logic for generating the error message I would still need to replicate that logic to differentiate between writing a partial failure status
https://github.com/elastic/kibana/pull/87489/files/80bfa5ac581ed5e320833f853ba9253577856b23#diff-333497786c31c4722a433be429554127c7eb40198062ef5004a057da5c0a23cfR86

and writing an error status
https://github.com/elastic/kibana/pull/87489/files/80bfa5ac581ed5e320833f853ba9253577856b23#diff-333497786c31c4722a433be429554127c7eb40198062ef5004a057da5c0a23cfR96

So I could get rid of the let either with a function to just do that or to by returning inside of the if/else if statements but this logic would still be replicated twice to do the status writing.